tracer/src/Datadog.Trace/Configuration/ConfigurationSources/CompositeConfigurationSource.cs

```diff
@@ -51,50 +53,260 @@ public void Add(IConfigurationSource source)
 
         /// <inheritdoc />
         public ConfigurationResult<string> GetString(string key, IConfigurationTelemetry telemetry, Func<string, bool>? validator, bool recordValue)
-            => _sources
-              .Select(source => source.GetString(key, telemetry, validator, recordValue))
-              .FirstOrDefault(value => value.IsValid, ConfigurationResult<string>.NotFound());
+        {
```

Comment:
The only difference in all these `GetXXX` is the value retrieving function, which could be a lambda, to avoid the n-plicated code